ci: enable windows 'sanitycheck --build-only'

thst-nordic opened the pull request:
- Small tweaks to fix paths/memory allocation.
- Use make from chocolatey

Signed-off-by: Thomas Stilwell <Thomas.Stilwell@nordicsemi.no>

thst-nordic commented on scripts/sanitycheck:
```diff
@@ -1404,7 +1398,7 @@ class MakeGenerator:
 
                 else:
-                    goal = self.goals[name]
+                    goal = self.goals[name.strip()]
```
extra windows line ending needs to be removed: '\r'

thst-nordic commented on scripts/sanitycheck:
```diff
@@ -1800,6 +1793,15 @@ class TestCase:
 
+            if os.name == 'nt':
+                with contextlib.closing(mmap.mmap(inf.fileno(), 0, access=mmap.ACCESS_READ)) as main_c:
+                    return mmap_worker(main_c)
```
[mmap on windows doesn't support MAP_PRIVATE. ](https://docs.python.org/3.0/library/mmap.html)

thst-nordic commented on scripts/sanitycheck:
```diff
@@ -3208,6 +3210,16 @@ def main():
+            print("Only build without size report works for now:")
+            print("\t- '--build-only --disable-size-report'")
+            print("\t- Requires make: 'choco install make'")
+            print("\t- May require long-path support in Windows OS.")
```
sample paths are under 256, but the test paths are mostly >256

carlescufi commented on scripts/sanitycheck:
```diff
@@ -1771,11 +1765,10 @@ class TestCase:
-        warnings = None
 
         with open(inf_name) as inf:
-            with contextlib.closing(mmap.mmap(inf.fileno(), 0, mmap.MAP_PRIVATE,
```
Wouldn't it be simpler to create a variable mmap_flags that you set conditionally for UNIX or Windows and then keep the code as-is without having to move it to a nested function and having to have 2 calls to mmap()?

thst-nordic replied:
yes, I'll update.

carlescufi commented on scripts/sanitycheck:
```diff
@@ -3208,6 +3210,16 @@ def main():
     options = parse_arguments()
 
+    if os.name == 'nt':
+        if (not options.build_only) or (not options.disable_size_report):
```
nitpicking, but why not have a single `if X and Y` instead of 2 `if` statements?

thst-nordic replied:
I can update it.  

print("Testing with sanitycheck on Windows is not fully supported yet.")
 ^^ was above the 2nd if before I moved it upstream.

thst-nordic committed:
ci: enable windows "sanitycheck --build-only"

- Small tweaks to fix paths/memory allocation

- Uses make from chocolatey

Signed-off-by: Thomas Stilwell <Thomas.Stilwell@nordicsemi.no>

carlescufi commented:
@nashif can you please take a look?

thst-nordic committed:
ci: skip file limit changes in sanitycheck when OS=Windows

the resource package installs on windows but fails on import and fails
    on the call :(

Signed-off-by: Thomas Stilwell <Thomas.Stilwell@nordicsemi.no>

marc-hb commented on scripts/sanitycheck:
```diff
@@ -3317,6 +3322,16 @@ def main():
+       (not options.build_only or not options.disable_size_report):
+        print("Testing with sanitycheck on Windows is not fully supported yet.")
+        print("Only build without size report works for now:")
+        print("\t- '--build-only --disable-size-report'")
```
I don't think burying the start of Windows documentation in just one catch-all Windows error message is a good idea. I think it's better and more "scalable" to keep Windows-specific tweaks separate from each other and generally closer to the corresponding code. This also allows "smarter" behaviour, here for instance.

```python
if os.name == 'nt' and not options.disable_size_report:
     warning ("Windows doesn't support size report yet, forcing --disable-size-report")
     options.disable_size_report = true
```
Just a suggestion/example of something not really possible with a grab bag error message.

marc-hb commented on scripts/sanitycheck:
```diff
@@ -3317,6 +3322,16 @@ def main():
+        print("Testing with sanitycheck on Windows is not fully supported yet.")
+        print("Only build without size report works for now:")
+        print("\t- '--build-only --disable-size-report'")
+        print("\t- Requires make: 'choco install make'")
```
I don't see why an installation tip for make is buried in an error message about unrelated options.

You already updated the documentation which is great. Now what is the error message for people who already installed their environment earlier and miss make? If it's something like "make not found" then there's nothing to do, you can just remove that line. If the error is more cryptic for some unknown reason then implement a better error message just before the first make invocation:

```python
if make_not_found:
    error("make not found, check the getting started guide")
```

marc-hb commented on scripts/sanitycheck:
```diff
@@ -3317,6 +3322,16 @@ def main():
+        print("Only build without size report works for now:")
+        print("\t- '--build-only --disable-size-report'")
+        print("\t- Requires make: 'choco install make'")
+        print("\t- May require long-path support in Windows OS.")
```
Similar comment: I don't see why this  information is tied to --build-only and --disable-size-report (and missing from issue #2664?)

marc-hb reviewed (changes requested):
Please reference issue #2664 in the commit message to enable more github magic and help people find more information.

thst-nordic commented:
closing due to refactor.

marc-hb commented:
> closing due to refactor.

For the record: PR #18603
